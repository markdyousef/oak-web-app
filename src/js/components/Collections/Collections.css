.container {
    width: 100%;
}

.toolbar {
    display: flex;
    width: 100%;
    justify-content: flex-end;
    padding: 20px;
}

.grid {
    width: 100%;
    padding: 20px;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.grid a {
    margin: 10px;
}
