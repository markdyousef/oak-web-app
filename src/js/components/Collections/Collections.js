// @flow
import React, { Component, PropTypes } from 'react';
import { Link } from 'react-router';
import styled from 'styled-components';
import colors from '../../styles/colors';
import CollectionCard from '../CollectionCard';
import Button from '../shared/Button';
import CollectionDialog from '../../containers/CollectionDialogContainer';

const Container = styled.div`
    width: 100%;
`;

const Header = styled.section`
    width: 100%;
    background-color: ${colors.white};
    padding: 48px 72px;
    border-bottom: 1px solid ${colors.lightGrey};
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
`;

const Info = styled.div`
    max-width: 500px;
    & h1 {
        font-size: 24px;
        font-weight: bold;
        padding-bottom: 24px;
        display: block;
        letter-spacing: -.04em;
    }
    & p {
        font-size: 18px;
        font-weight: normal;
        padding-bottom: 40px;
        display: block;
        line-height: 1.48;
    }
`;

const Stats = styled.div`
    & div {
        display: inline-block;
        margin-right: 20px;
    }
    & h3 {
        font-size: 20px;
        font-weight: bold;
        padding-bottom: 8px;
        display: block;
    }
    & h5 {
        font-size: 14px;
        font-weight: 300;
        text-transform: uppercase;
        padding-bottom: 8px;
    }
`;

const ButtonGroup = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    margin-top: 20px;
`;

const Grid = styled.section`
    width: 100%;
    padding: 20px;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    & a {
        margin: 8px;
    }
`;

class Collections extends Component {
    static propTypes = {
        data: PropTypes.shape({
            loading: PropTypes.bool,
            groves: PropTypes.arrayOf(PropTypes.shape({
                name: PropTypes.string.isRequired,
                id: PropTypes.string.isRequired,
                description: PropTypes.string
            }))
        }).isRequired
    };
    static defaultProps = {};
    constructor() {
        super();
        this.state = {
            showAdd: false,
            collections: 0,
            cards: 0,
            members: 0
        };
    }
    renderCollections = () => {
        const { data } = this.props;

        if (data.loading) return <div>LOADING</div>

        if (!data.loading && data.groves) {
            return (
                <Grid>
                    {data.groves.map(grove =>
                        <Link to={`/collection/${grove.id}`} key={grove.id}>
                            <CollectionCard name={grove.name} description={grove.description} />
                        </Link>
                    )}
                </Grid>
            );
        }

        return null;
    }
    render() {
        const { showAdd, collections, cards, members } = this.state;
        const { data } = this.props;
        return (
            <Container>
                <Header>
                    <Info>
                        <h1>Collections</h1>
                        <p>A collection of useful articles, work, notes & anything else needed to stimulate collective learning.</p>
                        <Stats>
                            <div>
                                <h3>{collections}</h3>
                                <h5>Collections</h5>
                            </div>
                            <div>
                                <h3>{cards}</h3>
                                <h5>Cards</h5>
                            </div>
                            <div>
                                <h3>{collections}</h3>
                                <h5>Teammates</h5>
                            </div>
                        </Stats>
                    </Info>
                    <ButtonGroup>
                        <Button
                            onClick={() => this.setState({ showAdd: true })}
                            text="Add Collection"
                            type="primary"
                        />
                        {showAdd &&
                            <CollectionDialog
                                close={() => {
                                    data.refetch();
                                    this.setState({ showAdd: false });
                                }}
                            />
                        }
                    </ButtonGroup>
                </Header>
                {this.renderCollections()}
            </Container>
        );
    }
}

export default Collections;
