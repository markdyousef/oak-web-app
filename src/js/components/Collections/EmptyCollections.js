// @flow
import React from 'react';
import styled from 'styled-components';
import Button from '../shared/Button';
import empty from '../../../img/collections-empty-state.svg';

const Empty = ({ onClick }:Object) => {
    return (
        <div>
            <img src={empty} alt="empty state" />
        </div>
    );
}

export default Empty;
