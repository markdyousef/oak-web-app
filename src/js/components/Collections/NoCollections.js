// @flow
import React from 'react';
import styled from 'styled-components';
import Button from '../shared/Button';
import empty from '../../../img/collections-empty-state.svg';

const Container = styled.div`
    width: 60%;
    max-width: 400px;
    margin: auto;
    margin-top: 30px;
    display: flex;
    flex-direction: column;
    align-items: center;

    & img {
        width: 60%;
    }
    & h2 {
        font-size: 20px;
        font-weight: lighter;
        text-align: center;
        text-transform: uppercase;
        margin-bottom: 40px;
    }
    & p {
        font-size: 15px;
        line-height: 24px;
        text-align: center;
        margin: 20px 0;
    }
`;

const Empty = ({ onClick }:Object) => {
    return (
        <Container>
            <h2>Collections</h2>
            <img src={empty} alt="empty state" />
            <p>You havenâ€™t created any collections yet. Create your first collection by hitting the button below.</p>
            <Button
                text="Create a card"
                onClick={onClick}
            />
        </Container>
    );
}

export default Empty;
