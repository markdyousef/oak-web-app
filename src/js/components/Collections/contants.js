// @flow


// ANALYTICS
export const CREATE_COLLECTION: string = 'COLLECTIONS/CREATE_COLLECTION';
