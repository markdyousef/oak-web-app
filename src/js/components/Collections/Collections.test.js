import React from 'react';
import renderer from 'react-test-renderer';
import Collections from './Collections';

it('renders correctly', () => {
    const props = {
        data: {}
    };
    const component = renderer.create(
        <Collections {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
