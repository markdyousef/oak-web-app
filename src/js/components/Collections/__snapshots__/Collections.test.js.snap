exports[`test renders correctly 1`] = `
<div
  className="gmuKOB">
  <section
    className="pjIGs">
    <div
      className="jtxdxF">
      <h1>
        Collections
      </h1>
      <p>
        A collection of useful articles, work, notes & anything else needed to stimulate collective learning.
      </p>
      <div
        className="dzrwBE">
        <div>
          <h3 />
          <h5>
            Collections
          </h5>
        </div>
      </div>
    </div>
    <div
      className="cWGcft">
      <button
        className="hQjvYS"
        onClick={[Function]}
        type="primary">
        Add Collection
      </button>
    </div>
  </section>
  <div
    className="gVkhVA">
    <h2>
      Collections
    </h2>
    <img
      alt="empty state"
      src="test-file-stub" />
    <p>
      You havenâ€™t created any collections yet. Create your first collection by hitting the button below.
    </p>
    <button
      className="jpCUHi"
      onClick={[Function]}
      type={undefined}>
      Create Collection
    </button>
  </div>
</div>
`;
